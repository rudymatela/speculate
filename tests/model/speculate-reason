WARNING: most laws below are not correct (see source)
_ :: Int  (holes: Int)
_ :: (Expr,Expr)  (holes: (Expr,Expr))
_ :: Expr  (holes: Expr)
_ :: Bool  (holes: Bool)
_ :: [(Expr,Expr)]  (holes: [(Expr,Expr)])
_ :: Thy  (holes: Thy)
okThy :: Thy -> Bool
insert :: (Expr,Expr) -> Thy -> Thy
complete :: Thy -> Thy
append :: Thy -> [(Expr,Expr)] -> Thy
normalize :: Thy -> Expr -> Expr
isNormal :: Thy -> Expr -> Bool
equivalent :: Thy -> Expr -> Expr -> Bool
initialize :: Int -> (Expr -> Expr -> Bool) -> [(Expr,Expr)] -> Thy
finalize :: Thy -> Thy
emptyThy :: Thy
True :: Bool
False :: Bool

 normalize emptyThy e == e
        normalize t e == normalize u e
              okThy t == True
  isNormal emptyThy e == True
         isNormal t e == isNormal u f
         append t eqs == t
complete (complete t) == complete t
finalize (finalize t) == finalize t

WARNING: most laws above are not correct (see source)
