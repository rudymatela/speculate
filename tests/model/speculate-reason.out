max expr size  =    4
  |- on ineqs  =    3
  |- on conds  =    3
max  #-tests   = 6000
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Bool  (holes: Bool)
_ :: Equation  (holes: Equation)
_ :: Expr  (holes: Expr)
_ :: Thyght  (holes: Thyght)
_ :: [Expr]  (holes: [Expr])
okThy :: Thyght -> Bool
insert :: Expr -> Thyght -> Thyght
complete :: Thyght -> Thyght
normalize :: Thyght -> Expr -> Expr
equivalent :: Thyght -> Expr -> Expr -> Bool
append :: Thyght -> [Expr] -> Thyght
emptyThy :: Thyght
True :: Bool
False :: Bool
Equation :: Expr -> Expr -> Equation

              okThy t == True
     equivalent t e e == True
     equivalent t e f == equivalent t f e
 normalize emptyThy e == e
complete (complete t) == complete t

